src/realm/util/file.cpp

```diff
@@ -1625,6 +1676,12 @@ File::UniqueID File::get_unique_id(FileDesc file)
     REALM_ASSERT(file >= 0);
     struct stat statbuf;
     if (::fstat(file, &statbuf) == 0) {
+        // On exFAT systems the inode and device are not populated
+        // until the file has been allocated some space. This has led
+        // to bugs where a unique id returned here was reused by different
+        // files. The following assertion ensures that this is not happening
+        // anywhere else in future code.
+        REALM_ASSERT_EX(statbuf.st_size > 0, statbuf.st_size, file);
```

Comment:
This should probably be a runtime error instead of an assertion. While the most likely reason for the file being empty is a bug in our code, it could also be due to someone else truncating it immediately before this function is called.